util/network-devp2p/src/node_table.rs

```diff
@@ -122,18 +123,31 @@ impl FromStr for NodeEndpoint {
 				address: a,
 				udp_port: a.port()
 			}),
-			Ok(_) => Err(ErrorKind::AddressResolve(None).into()),
-			Err(e) => Err(ErrorKind::AddressResolve(Some(e)).into())
+			Ok(None) => {
+				// REVIEW: how is this case possible? I think we can remove this case.
+				Err(ErrorKind::AddressResolve(None).into())
```

Comment:
I think we shouldn't remote it, cause we cannot prove that std api, will not change underlaying implementation.

anyway, I think it's also more idiomatic to use `bail!` macro in cases like this one, e.g.

```rust
Ok(None) => bail!(ErrorKind::AddressResolve(None)),
```

Reply:
Can you elaborate on this a little? Not sure I understand what you mean by "shouldn't remote it". Thanks!

Reply:
Oh, I think you meant to type "shouldn't REMOVE" it. Now I get it!